rivers.py

```
@@ -0,0 +1,302 @@
+            point = QgsPointXY(feature['start_x'], feature['start_y'])
+            new_feature = QgsFeature()
+            new_feature.setFields(fields)
+            new_feature.setGeometry(QgsGeometry.fromPointXY(point))
+            new_feature['x'] = feature['start_x']
+            new_feature['y'] = feature['start_y']
+            new_feature['z'] = feature['start_z']
+            pointLayer.addFeature(new_feature)
+
+        elif feature['end_x'] != None and feature['end_y'] != None and feature['end_z'] != None and feature['end_z'] == feature['max_z']:
+            point = QgsPointXY(feature['end_x'], feature['end_y'])
+            new_feature = QgsFeature()
```

Ruff (0.7.0)

Undefined name `QgsFeature`

(F821)